apis/kueue/v1beta1/workload_types.go

```diff
@@ -365,6 +365,38 @@ type WorkloadStatus struct {
 	//
 	// +optional
 	AccumulatedPastExexcutionTimeSeconds *int32 `json:"accumulatedPastExexcutionTimeSeconds,omitempty"`
+
+	// waitForPodsReadyRequeuesByReason holds the re-queue state for WaitForPodsReady.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=reason
+	// +patchStrategy=merge
+	// +patchMergeKey=reason
+	WaitForPodsReadyRequeuesByReason []WaitForPodsReadyRequeue `json:"waitForPodsReadyRequeuesByReason,omitempty"`
```

Comment:
I understand what is the objective for this field. In that case, why can we not reuse `requeueState` field?
Why can we not just add reason field to `requeueState` field?
Why do we need to store both `StartupTimeout` and `RecoveryTimeout` reasons?

Reply:
We probably could, but we would need to extend it, because currently `requeueState` does not hold the reason at all. Moreover, `requeueState` is incremented for other reasons too, such as Failed ProvisiningRequest.

I opened the issue on behalf of a user, and in the user request they want a metric only specifically for WaitForPodsReady.

Let us know if you have some specific idea how to extend `requeueState` to include the necessary information to build the metric.

Reply:
+1 to what @mimowo has said, I think it would be useful to track the reason behind eviction/requeueing as well, not only the fact it was requeued